packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.js

```diff
@@ -146,7 +146,8 @@ async function serviceWalkthrough(context, defaultValuesFilename, serviceMetadat
   // During API add, make sure we're creating a transform.conf.json file with the latest version the CLI supports.
   await updateTransformerConfigVersion(resourceDir);
 
-  if (resolverConfig.project || resolverConfig.models) {
+  // resolverConfig exists and it has a model or project sync config write it to the transformer
+  if (resolverConfig && (resolverConfig.project || resolverConfig.models)) {
```

Comment:
Cant we have this validation inside `writeResolverConfig` instead of having these check here

Reply:
moved check inside `writeResolverConfig`